### ISSUE TITLE
[request] Allow enum overloads, without unnecesarily adding `enumval` to their names

### ISSUE THREAD

**Issue by heysokam** [PMunch/futhark#95]
Quote from discord:
> Basically this line has to be done without the global state:
> https://github.com/PMunch/futhark/blob/master/src/futhark.nim#L349
> Since `usedNames` is passed in (through state, but there is an overload here) it should be fairly simple to create a new one for each enum and just use `sanitizeName` with that

**Comment by PMunch** [comment]
Tried to implement this but Nim wasn't happy with the redefinition either.. Could you make a test case showcasing what you wanted?

**Comment by heysokam** [comment]
> Tried to implement this but Nim wasn't happy with the redefinition either.. Could you make a test case showcasing what you wanted?

You are right. I just tested without a `{.pure.}` enum and nim is not happy :(
I guess I only tried this with pure before? I thought enum fields didn't collide with any other symbol :thinking:
```nim
type Thing *{.pure.}= enum
  One, Two

type One * = object
  val *:int

echo One(val: 1)
echo $One
```This works, but apparently it doesn't without pure

**Comment by PMunch** [comment]
Hmm, I have to read up on what `pure` does. It used to be a lot stricter, but if it's relaxed enough now I'd be open to marking all enums as pure. Another difficulty wa that the `when declared` checks tripped up. So I'll need to refine those in a way which makes them only fire if it would trigger a collision.
